app/assets/javascripts/images/share_modal.js

```
@@ -0,0 +1,50 @@
+import $ from 'jquery';
+
+const UNPROCESSABLE_ENTITY = 422;
+const NOT_FOUND = 404;
+
+function setFlashMessage(type, message) {
+  document.getElementById('flash_messages').innerHTML =
+    `<div class="alert alert-${type} %>">${message}</div>`;
+}
+
+class SharerModal {
+  constructor(modalSelector) {
+    this.$modal = $(modalSelector);
+  }
+
+  setupModal({imageId, imageUrl}) {
+    this.$modal.find('img').attr('src', imageUrl);
+    const formAction = this.$modal.find('form').attr('action');
+    this.$modal.find('form').attr('action', formAction.replace(':id', imageId));
+  }
+
+  attachEventHandlers() {
+    this.$modal.on('show.bs.modal', event => {
+      const modalTrigger = event.relatedTarget;
+      this.setupModal(modalTrigger.dataset);
+    });
+    this.$modal.on('hidden.bs.modal', function() {
+      console.log("hi");
+      $('.form-group').find("input,text,textarea").val('').end();
+      $('.form-group').find('span.help-block').hide();
+      $('.form-group').removeClass('has-error');
```

Let's take a step back from the code and focus on the general approach.

Right now, you're basically saying "on modal hide, clear all the input values, hide the error message, and remove the error styles".

Can you think of another way to solve this problem that's both simpler and less brittle? (Again, think about about a general approach rather than any code implementation right now.)